m4/ul.m4

```diff
@@ -590,3 +590,33 @@ AC_DEFUN([UL_TINFO_CHECK], [
        TINFO_CFLAGS=""])
   ])
 ])
+
+AC_DEFUN([_UL_SIZEOF_TIME_T], [
+  AC_COMPUTE_INT($1, [sizeof(time_t)], [[
+    #include <time.h>
+    #include <stdio.h>
+  ]])
+])
+
+dnl
+dnl UL_SYS_TIME64
+dnl
+dnl Ensure 64bit times are used.
+dnl
+AC_DEFUN([UL_SYS_TIME64], [
+  AC_MSG_CHECKING([checking flags necessary for 64bit time_t])
+  _UL_SIZEOF_TIME_T(ul_sizeof_time_t)
+
+  if test "$ul_sizeof_time_t" -lt 8; then
+    AC_DEFINE([_TIME_BITS], [64], [Ensure 64bit time values])
+
+    _UL_SIZEOF_TIME_T(ul_sizeof_time_t_new)
+    if test "$ul_sizeof_time_t_new" -lt 8; then
+      AC_MSG_RESULT([not possible])
+      AC_MSG_WARN([Could not make time_t 64bits wide])
+    fi
+    AC_MSG_RESULT([-D_TIME_BITS=64])
```

Comment:
Note that glibc needs LFS as well for time64 and `_TIME_BITS` is a glibcism (i.e. for this case, you always need LFS).

Reply:
`AC_SYS_LARGEFILE` is used in `configure.ac` before `UL_SYS_TIME64`.